code/addons/test/src/manager.tsx

```diff
@@ -73,64 +73,75 @@ addons.register(ADDON_ID, (api) => {
       },
 
       stateUpdater: (state, update) => {
-        if (!update.details?.testResults) {
-          return;
+        const updated = {
+          ...state,
+          ...update,
+          details: { ...state.details, ...update.details },
+        };
+
+        if ((!state.running && update.running) || (!state.watching && update.watching)) {
+          // Clear coverage data when starting test run or enabling watch mode
+          delete updated.details.coverageSummary;
         }
```

Comment:
I'm somewhat skeptical if this is the proper solution to this problem.

Would an emit from node with an empty `details.coverageSummary` be a cleaner/simpler solution?

Reply:
That would involve more moving parts, so it's not my preference. We can reconsider when we implement server-led synchronized state management.